provider/http.go

```diff
@@ -14,15 +14,15 @@ import (
 type HTTPClient struct {
 	pb.UnimplementedApiServer
 
-	baseURL    string
-	httpClient *http.Client
-	requestID  utils.RequestID
-	privateKey *solana.PrivateKey
-	authHeader string
+	BaseURL    string
+	HttpClient *http.Client
+	RequestID  utils.RequestID
+	PrivateKey *solana.PrivateKey
+	AuthHeader string
 }
 
 func (h *HTTPClient) GetAuthHeader() string {
```

Comment:
nit: should be called `AuthHeader()`, or can be deleted entirely if we need to make the properties public